packages/backend/server/src/core/quota/override.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { FeatureType } from '../features';
+import type { QuotaBusinessType } from './types';
+
+export abstract class QuotaOverride {
+  abstract readonly name: string;
+  abstract overrideQuota(
+    ownerId: string,
+    workspaceId: string,
+    features: FeatureType[],
+    quota: QuotaBusinessType
+  ): Promise<QuotaBusinessType>;
+}
+
+@Injectable()
+export class QuotaOverrideService {
```

Comment:
there is no need to provide such complex way to update a quota. just expose a `updateQuota` with type guard